internal/dot/graph.go

```diff
@@ -25,15 +25,36 @@ import (
 	"reflect"
 )
 
+// FailureState of a constructor or group is updated when they fail to build.
+type FailureState int
```

Comment:
This conflicts with @abhinav's comment, but seeing as `success` is a valid `FailureState`, we should consider renaming to just `State`. The different states could then be:

```
const (
  success State = iota
  transitiveFailure
  rootFailure
)
```

@abhinav + @leyao What do you think? I'm fine with either way.

Reply:
Fair point, although there is precedence for "zero value is different" when it
comes to iota. For example, in yarpcerrors, [CodeOK] means "no error" even
though `Code` represents the error code.

We could rename this to `ErrorType` and `success` to `noError`, which might
make more sense as a zero value then, but I don't feel strongly about it.
I'm open to other names too. I wanted to avoid `State` because it's pretty
generic on its own. `Ctor.State` doesn't tell you what kind of state we're
talking about. The `Ctor` struct is the state of the Ctor already.

Reply:
Yea that's a good point. I'd lean towards your `ErrorType` suggestion in this case.

Reply:
`ErrorType` sounds good, I'll go with that then 👍 Also updated `failed` to `transitiveFailure` for clarity and to match `ErrorType`.